File: Leetcode/cpp/sol3232.cpp
//https://leetcode.com/problems/find-if-digit-game-can-be-won/description/

class Solution {
public:
    bool canAliceWin(vector<int>& nums) {

        for(auto i = 0; i<nums.size(); i++){
            if (nums[i]<10){
                nums[i] = - nums[i];
            }
        }

        return reduce(nums.begin(), nums.end(), 0);
    }
};

File: Leetcode/python/sol3232.py
# https://leetcode.com/problems/find-if-digit-game-can-be-won/description/

class Solution:
    def canAliceWin(self, nums: List[int]) -> bool:
        result: int = 0
        for i in nums:
            if i < 10:
                result+=i
            else:
                result-=i
        return result != 0
